### DISCUSSION TITLE
Push changes in postgis realtime to maplibre

### DISCUSSION THREAD

**Discussion by Fabioni** [maplibre/martin#1359] [Q&A]
Hey, I want to use Martin to show vector tiles in maplibre from a data source in postgis.

The data in postgis changes frequently and I want the changes immediately shown on client side. 

The only method I found so far is having a seperate websocket connection to notify the client of any change and then invalidating the complet vector map layer in maplibre.

Do you know any better alternative with only differential updating or at least only updating the tiles that have updates? Is there any functionality in Martin I can use for that?

**Comment by nyurik** [comment]
Martin doesn't have any way to notify the client of data invalidation - mostly because we do not have a standard way to do that in MapLibre JS or Native clients. If we can start this discussion at https://github.com/maplibre/maplibre-gl-js and there is a buy-in on the overall approach (maybe as a plugin at first), then we can implement support for that here as well.

**Reply by Fabioni** [reply]
so what would be the current best approach in your opinion?

**Reply by nyurik** [reply]
I think what you are doing is workable. If you propose it in an issue per above link, the community will give a better answer with respect to the best client side options.
